MatrixSDK/Utils/Profiling/MXTaskProfile.h

```diff
@@ -18,13 +18,28 @@
 
 NS_ASSUME_NONNULL_BEGIN
 
+typedef NSString *const MXTaskProfileName NS_TYPED_EXTENSIBLE_ENUM;
```

Comment:
Another possibility:
```
extern const struct MXTaskProfileName {
    const struct Startup {
        NSString* __unsafe_unretained _nonnull_ Category; // Category name
        NSString* __unsafe_unretained _nonnull_ InitialSync;
        NSString* __unsafe_unretained _nonnull_ IncrementalSync;
    } Startup;    
    const struct PushNotifications {
        NSString* __unsafe_unretained _nonnull_ Category; // Category name
        NSString* __unsafe_unretained _nonnull_ OpenEvent;
    } PushNotifications;
} MXTaskProfileName;
```

Reply:
In terms of using the above, I've had a quick look and I get as far as:

```objc
- (MXTaskProfile *)startMeasuringTaskWithName:(struct MXTaskProfileName)name;
```
```objc
MXTaskProfile *taskProfile = [MXSDKOptions.sharedInstance.profiler startMeasuringTaskWithName:MXTaskProfileName.Startup.InitialSync];
```
However then it complains 
> Sending 'NSString *__unsafe_unretained const __unsafe_unretained' to parameter of incompatible type 'struct MXTaskProfileName'

which I assume means that `startMeasuringTaskWithName` would then need to accept a plain string again?